sql/07-bde_replication.sql

```diff
@@ -0,0 +1,45 @@
+--------------------------------------------------------------------------------
+--
+-- linz-bde-schema
+--
+-- Copyright 2019 Crown copyright (c)
+-- Land Information New Zealand and the New Zealand Government.
+-- All rights reserved
+--
+-- This software is released under the terms of the new BSD license. See the
+-- LICENSE file for more information.
+--
+--------------------------------------------------------------------------------
+-- Create publication for tables in bde. schema
+--------------------------------------------------------------------------------
+
+DO $PUBLICATION$
+DECLARE
+    rec RECORD;
+BEGIN
+
+    IF NOT EXISTS ( SELECT true FROM pg_class c, pg_namespace n
+                WHERE n.nspname = 'pg_catalog'
+                AND c.relnamespace = n.oid
+                AND c.relname = 'pg_publication' )
+    THEN
+        -- This version of PostgreSQL
+        -- does not support logical replication
+        RETURN;
+    END IF;
+
+    -- NOTE: we're using EXECUTE here to skip run for PostgreSQL
+    --       not supporting 'PUBLICATION' syntax (< 10)
```

Comment:
Should we rather just check for > PG 10 in this function, or at least catch the exception?

Reply:
I've checked that PostgreSQL 9.6 does not have the pg_publication table, so the check should be equivalent and not require a version parsing routine. I don't expect any exception (tested against 9.6 and 10)